### ISSUE TITLE
微调分类任务出现零损失

### ISSUE THREAD

**Issue by kabugge** [modelscope/ms-swift#6845]
你好，
我在微调Qwen 2.5 Omni 7B模型，下面是一条训练数据：
```
{
    "messages": [
      {
        "content": "<video>背景：作为一名专业的视频评价员，对这条视频进行评估。XXX这里省略",
        "role": "user"
      }
    ],
    "label": 2,
    "videos": [
      "xxx.mp4"
    ]
  }
```
标签有0，1，2。微调的过程中损失一直是0.0，是哪里出问题了吗？

**Comment by Jintao-Huang** [comment]
看看 grad_norm是什么，训练脚本发一下

**Comment by kabugge** [comment]
Hi @Jintao-Huang 
训练脚本是这样：
swift sft     --model Qwen/Qwen2.5-Omni-7B     --task_type seq_cls     --num_labels 3     --problem_type single_label_classification     --train_type lora     --dataset data/bz.json     --split_dataset_ratio 0.1     --eval_strategy steps     --torch_dtype bfloat16     --num_train_epochs 3 --per_device_train_batch_size 1     --per_device_eval_batch_size 1     --learning_rate 1e-4     --lora_rank 32     --lora_alpha 64     --target_modules all-linear     --gradient_accumulation_steps 16     --eval_steps 50     --save_steps 50     --save_total_limit 50     --logging_steps 5     --max_length 5120     --output_dir output     --system 'You are a helpful assistant.'     --warmup_ratio 0.05     --dataloader_num_workers 4     --model_author swift     --model_name swift-robot

我加上--task_type seq_cls     --num_labels 3     --problem_type single_label_classification后，训练loss就正常了，之前没这三个参数。但现在验证损失比训练损失小10倍。。。感觉不太对。。。

以下是一段训练输出：
{'loss': 0.72207642, 'grad_norm': 82.58652496, 'learning_rate': 5.263e-05, 'acc': 0.6875, 'epoch': 1.53, 'global_step/max_steps': '240/471', 'percentage': '50.96%', 'elapsed_time': '2h 40m 38s', 'remaining_time': '2h 34m 37s', 'memory(GiB)': 27.47, 'train_speed(iter/s)': 0.024899}
{'loss': 0.64030762, 'grad_norm': 96.23352814, 'learning_rate': 5.088e-05, 'acc': 0.7625, 'epoch': 1.56, 'global_step/max_steps': '245/471', 'percentage': '52.02%', 'elapsed_time': '2h 43m 39s', 'remaining_time': '2h 30m 58s', 'memory(GiB)': 27.47, 'train_speed(iter/s)': 0.024949}
{'loss': 0.7442688, 'grad_norm': 41.88717651, 'learning_rate': 4.912e-05, 'acc': 0.7, 'epoch': 1.59, 'global_step/max_steps': '250/471', 'percentage': '53.08%', 'elapsed_time': '2h 46m 44s', 'remaining_time': '2h 27m 23s', 'memory(GiB)': 27.47, 'train_speed(iter/s)': 0.02499}
Train:  53%|███████████████████████████████████                               | 250/471 [2:46:44<2:16:29, 37.06s/it]
{'eval_loss': 0.05101554, 'eval_runtime': 199.9298, 'eval_samples_per_second': 1.395, 'eval_steps_per_second': 1.395, 'eval_acc': 0.65591398, 'epoch': 1.59, 'global_step/max_steps': '250/471', 'percentage': '53.08%', 'elapsed_time': '2h 50m 4s', 'remaining_time': '2h 30m 20s', 'memory(GiB)': 27.47, 'train_speed(iter/s)': 0.024499}
Val: 100%|████████████████████████████████████████████████████████████████████████| 279/279 [03:17<00:00,  1.41it/s]
[INFO:swift] Saving model checkpoint to /root/ms-swift/output/v61-20251202-151219/checkpoint-250
{'loss': 0.75992432, 'grad_norm': 107.90041351, 'learning_rate': 4.737e-05, 'acc': 0.7125, 'epoch': 1.62, 'global_step/max_steps': '255/471', 'percentage': '54.14%', 'elapsed_time': '2h 53m 6s', 'remaining_time': '2h 26m 38s', 'memory(GiB)': 27.47, 'train_speed(iter/s)': 0.02455}
{'loss': 0.87790985, 'grad_norm': 90.72616577, 'learning_rate': 4.561e-05, 'acc': 0.625, 'epoch': 1.66, 'global_step/max_steps': '260/471', 'percentage': '55.20%', 'elapsed_time': '2h 56m 11s', 'remaining_time': '2h 22m 59s', 'memory(GiB)': 27.47, 'train_speed(iter/s)': 0.024594}

grad_norm通常在10-110之间。

**Comment by Jintao-Huang** [comment]
ms-swift/ transformers 版本是什么

**Comment by kabugge** [comment]
> ms-swift/ transformers 版本是什么
ms-swift                      3.10.0.dev0
transformers                  4.57.1

另外，--enable_channel_loss true似乎不起作用

**Comment by Jintao-Huang** [comment]
序列分类任务 没有加channel loss

**Comment by Jintao-Huang** [comment]
试试3.10.3版本, release/3.10

**Comment by kabugge** [comment]
> 试试3.10.3版本, release/3.10

好的，直接pip --upgrade install ms-swift==3.10.3吗？
序列分类任务可以加上channel loss吗？也可以告诉我，我自己试试加。。。

**Comment by kabugge** [comment]
> 试试3.10.3版本, release/3.10

这个版本装上了，验证损失仍然比训练损失小10倍。。
